tests/Aggregation/DateHistogramTest.php

```diff
@@ -18,7 +18,62 @@ class DateHistogramTest extends BaseAggregationTest
      */
     public function testDateHistogramAggregation(): void
     {
-        $agg = new DateHistogram('hist', 'created', '1h');
+        $this->_checkVersion('7.2');
+
+        $agg = new DateHistogram('hist', 'created');
+        $agg->setFixedInterval('1h');
```

Comment:
```suggestion
        $version = $this->_getVersion();

        if (\version_compare($version, '7.2') < 0) {
            $this->expectDeprecation('Since ruflin/elastica 7.1.0: The "Elastica\Aggregation\DateHistogram::setInterval()" method is deprecated, use "setDateInterval()" or "setCalendarInterval()" instead. It will be removed in 8.0.');

            $agg = (new DateHistogram('hist', 'created'))
                ->setInterval('1h')
            ;
        } else {
            $agg = (new DateHistogram('hist', 'created'))
                ->setFixedInterval('1h')
            ;
        }
```

Cannot we use a code like this to test on version lower than 7.2?

Reply:
```suggestion
        $version = $this->_getVersion();

        if (\version_compare($version, '7.2') < 0) {
            $agg = (new DateHistogram('hist', 'created'))
                ->setParam('interval', '1h')
            ;
        } else {
            $agg = (new DateHistogram('hist', 'created'))
                ->setFixedInterval('1h')
            ;
        }
```

Or like this?

Reply:
Maybe the deprecation is a little too violent, as this can't be fixed for versions lower than 7.2 unless using `setParam()` directly? WDYT @ruflin @thePanz ? Is there any advices/hints for this kind of deprecations?

Not sure if it's a blocker as the deprecation is also very helpful for upper compatibility.

Reply:
I like having deprecations, and the first example (where we test the deprecation) if :+1:  for me :)